### ISSUE TITLE
x64: bug in decoding instructions on 8-bit registers

### ISSUE THREAD

**Issue by LRGH** [bdcht/amoco#49]
``` python
>>> from amoco.arch.x64 import cpu_x64
>>> str(cpu_x64.disassemble('\x80\xcc\x0c'))
'or          spl, 0xc'
>>> str(cpu_x64.disassemble('\x40\x80\xcc\x0c'))
'or          spl, 0xc'
```

The first should be `or ah, 0xc`.

I am looking for how to correct this bug.
The function `getreg` in `env.py` never returns `ah`, therefore one solution would be to change this function to add a third argument (the value of REX) ; another possibility is to change `getModRM` in `utils.py`. It depends on if we need also to change `getregR`, `getregRW` and `getregB`.

My proposal for a patch is

``` python
--- a/amoco/arch/x64/utils.py
+++ b/amoco/arch/x64/utils.py
@@ -67,6 +67,16 @@ def getModRM(obj,Mod,RM,data):
     # r/16/32 case:
     if Mod==0b11:
         op1 = env.getreg((B<<3)+RM,opdsz)
+        if REX is None and opdsz is 8: op1 = {
+            env.al:  env.al,
+            env.cl:  env.cl,
+            env.dl:  env.dl,
+            env.bl:  env.bl,
+            env.spl: env.ah,
+            env.bpl: env.ch,
+            env.sil: env.dh,
+            env.dil: env.bh,
+            }[op1]
         return op1,data
     # m/16/32 case:
     if adrsz!=16 and RM==0b100:
```

**Comment by bdcht** [comment]
should be fixed as well once x64's getModRM function is rewritten...thanks for reporting these examples!

**Comment by bdcht** [comment]
fixed by a6535e5 (v2.4.5)
